pkg/artifactory/datasource_artifactory_file_test.go

```diff
@@ -44,14 +45,25 @@ func TestVerifySha256Checksum(t *testing.T) {
 func CreateTempFile(content string) (f *os.File, err error) {
 	file, err := ioutil.TempFile(os.TempDir(), "terraform-provider-artifactory-")
 
+	if err != nil {
+		return nil, err
+	}
+
+	if file == nil {
```

Comment:
Isn't `ioutil.TempFile` guaranteed to return a file if `err` is `nil`? Did you run into a case that it didn't?